typings/pluginapi.d.ts

```diff
@@ -0,0 +1,183 @@
+/**
+ * This file describes the code-server plugin API for adding new applications.
+ */
+import { Logger } from "@coder/logger"
+import * as express from "express"
+
+/**
+ * Overlay
+ *
+ * The homepage of code-server will launch into VS Code. However, there will be an overlay
+ * button that when clicked, will show all available applications with their names,
+ * icons and provider plugins. When one clicks on an app's icon, they will be directed
+ * to <code-server-root>/<plugin-path>/<app-path> to access the application.
+ */
+
+/**
+ * Plugins
+ *
+ * Plugins are just node modules.
+ *
+ * 1. code-server uses $CS_PLUGIN to find plugins.
+ *
+ * e.g. CS_PLUGIN=/tmp/will:/tmp/teffen will cause code-server to load
+ * /tmp/will and /tmp/teffen as plugins.
+ *
+ * 2. code-server uses $CS_PLUGIN_PATH to find plugins. Each subdirectory in
+ * $CS_PLUGIN_PATH with a package.json where the engine is code-server is
+ * a valid plugin.
+ *
+ * e.g. CS_PLUGIN_PATH=/tmp/nhooyr:/tmp/ash will cause code-server to search
+ * /tmp/nhooyr and then /tmp/ash for plugins.
+ *
+ * CS_PLUGIN_PATH defaults to
+ * ~/.local/share/code-server/plugins:/usr/share/code-server/plugins
+ * if unset.
+ *
+ *
+ * 3. Built in plugins are loaded from __dirname/../plugins
+ *
+ * Plugins are required as soon as they are found and then initialized.
+ * See the Plugin interface for details.
+ *
+ * If two plugins are found with the exact same name, then code-server will
+ * use the first one and emit a warning.
+ *
+ */
+
+/* Programmability
```

Comment:
To match the other blocks:
```suggestion
/**
 * Programmability
```

Reply:
Embarrasing for me tbh